specs/grafana_event_handling_in_monitoring_integration.adoc

```diff
@@ -0,0 +1,313 @@
+:imagesdir: ./images
+
+= Grafana Event Handling In Monitoring Integration
+
+Alerting system is essential to spot the problem and notify it to the corresponding user.
+The threshholds for alerts will be configured in grafana and the alerts will be raised by grafana.
+These alerts are captured by monitoring-integration module and notified to respective user via the notifier.
+
+
+== Problem description
+
+Grafana has been introduced to display interactive graphs for monitoring data
+for clusters and hosts (https://github.com/Tendrl/specifications/issues/168)
+and all the alerts henceforth will be configured in grafana. So an additional
+component needs to be introduced which would receive the alerts from grafana,
+and use the existing alerting system(Notifier) to notify respective users.
+
+== Use Cases
+
+Goal: Notify user in case of any alert event.
+
+Actors:
+
+1) User 
+
+Steps:
+
+* An alert is configured in grafana as per user requirement.
+
+* A Notification channel is configured in grafana.
+
+* Any change in system state is caught by tendrl component
+  via alerts sent by grafana.
+
+* The alert(Event for Tendrl) is converted into current alert 
+  structure and passed to node agent socket.
+
+* The new alert is sent to respective user via Notifier.
+
+
+== Proposed change
+
+
+image::alert_component.png[Tendrl Alerting System]
+
+
+Alert-Handler: It will receive alerts from Grafana and process the alerts.
+Notifier: Its function is to send notification to the users.(The present alert system)
+
+Alert Handler will expose an endpoint(webhook receiver) to receive alerts from grafana.
+This endpoint will be configured in grafana as a notification channel(This configuration
+will be done by Monitoring-Integration component).
+For now the end point should be at localhost.
+
+When alert-handler starts then the end point will be ready to receive
+the alerts from grafana. If any alert is raised then grafana will send the
+alert to the end point. After receiving the alert, it should be converted to
+current event structure of tendrl. Received grafana alert has very limited
+details so other details such as updated_at needs to fetched from grafana.
+This can be done in 2 ways:
```

Comment:
I wanted these alternatives to be the part of flow, else it would have been a tough to understand.
They are just an alternative sub-flow and not a complete new flow.
still if you think it should be a part of alternative then I will do it as required.